src/cryptonote_core/tx_pool.h

```diff
@@ -264,10 +264,11 @@ namespace cryptonote
      *
      * @param tx_infos return-by-reference the transactions' information
      * @param key_image_infos return-by-reference the spent key images' information
+     * @param show_privacy_critical_data show data that may be critical to the node privacy
      *
      * @return true
      */
-    bool get_transactions_and_spent_keys_info(std::vector<tx_info>& tx_infos, std::vector<spent_key_image_info>& key_image_infos) const;
+    bool get_transactions_and_spent_keys_info(std::vector<tx_info>& tx_infos, std::vector<spent_key_image_info>& key_image_infos, bool show_privacy_critical_data = true) const;
```

Comment:
Should this default to false?

Reply:
In terms of minimizing changes to current code, this should default to true (everything is shown currently, no restriction when in restricted-rpc, so defaulting to true is "backward compatible"). In terms of being on the safe side of maintaining privacy for future developments, should default to false.

Is this what you meant or how you see it?

Reply:
It seems better to me to favor being on the safe side over minimizing code changes: i.e. set default to `false` and explicitly pass `true` to all instances of calling this function (looks like there's only one instance in cryptonote_core.cpp).

Reply:
I agree and will update accordingly. Currently generalizing the approach so that every single call that makes use of sensitive information does not count the information-to-hide in it. (Some of them do implicitly like get_pool_transaction_stats). In practice means to integrate the filtering at low level (DB functions like `for_all_pool_txes`) rather than high level.
Will go broader than this particular issue, but very useful IMO.

Reply:
Oooh, good thinking, thanks.